File: core/spells/serialization/SerializableWeaving.cs
using System;
using System.Collections.Generic;

namespace Rzeka
{
  public struct SerializableWeaving : ISerlializableBindingSpell
    {
        public Guid guid { get; set; }
        public string title { get; set; }
        public SpellSchool spellSchool { get; set; }
        public string whosName { get; set; }
        public bool hasMana { get; set; }
        public Who Who { get; set; }
        public Dictionary<string, bool> ingredients { get; set; }
    }
}

File: ui/src/eris/types/common/spells/SerializableWeaving.ts
import { SpellSchool } from "./SpellSchoolEnum"
import { Dictionary } from "../Dictionary"
import { Who } from "../Who"
import { ISerlializableBindingSpell } from "./ISerlializableBindingSpell"

export class SerializableWeaving implements ISerlializableBindingSpell {
    Who: Who
    hasMana: boolean
    ingredients: Dictionary<Boolean>
    guid: string
    title: string
    spellSchool: SpellSchool
    whosName: string
    wasCast: boolean
}
